File: MedLabO.Models/Requests/ChangePasswordRequest.cs

namespace MedLabO.Models.Requests
{
    public class ChangePasswordRequest
    {
        public Guid UserId { get; set; }
        public string? OldPassword { get; set; }
        public string? NewPassword { get; set; }
        public string? ConfirmNewPassword { get; set; }
    }
}

File: MedLabO UI/medlabo_desktop/lib/models/common/change_password_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'change_password_request.g.dart';

@JsonSerializable()
class ChangePasswordRequest {
  String? userId;
  String? oldPassword;
  String? newPassword;
  String? confirmNewPassword;

  ChangePasswordRequest(
      {this.userId,
      this.oldPassword,
      this.newPassword,
      this.confirmNewPassword});

  factory ChangePasswordRequest.fromJson(Map<String, dynamic> json) =>
      _$ChangePasswordRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ChangePasswordRequestToJson(this);
}
